deployments/helm/nvidia-device-plugin/templates/daemonset-gfd.yml

```
@@ -199,6 +201,10 @@ spec:
         - name: host-sys
           hostPath:
             path: "/sys"
+        - name: imex-nodes-config-file
+          type: DirectoryOrCreate
+          hostPath:
+            path: {{ quote (include "nvidia-device-plugin.filepathJoin" (list .Values.nvidiaDriverRoot "etc/nvidia-imex") | clean) }}
```

```suggestion
        - name: nvidia-imex-dir
          type: DirectoryOrCreate
          hostPath:
            path: {{ quote (include "nvidia-device-plugin.filepathJoin" (list .Values.nvidiaDriverRoot "etc/nvidia-imex") | clean) }}
```

Can we do the quote and the clean inside the helper?

---

Also, I would think we want the path to have three elements `(list / .Values.nvidiaDriverRoot "etc/nvidia-imex")`. Otherwise if `.Values.nvidiaDriverRoot` does not begin with a slash, we will end up with a relative path, not an absolute one.

---

done

---

edited
i have tested
- null
- /proc/nvidia
- /proc/nvidia/
- proc/nvidia